src/DotNetEnv/Parsers.cs

```diff
@@ -173,17 +174,25 @@ private static string ToUtf32Char (string hex)
                 .Or(OctalChar)
                 .Or(EscapedChar);
 
+        private static readonly Parser<string> InlineWhitespace =
+            InlineWhitespaceChars.Many().Text();
+
         // unquoted values can have interpolated variables,
         // but only inline whitespace -- until a comment,
         // and no escaped chars, nor byte code chars
-        // FIXME: would be nice to solve multi word with comment case with parser directly (no split + trim)
+        private static readonly Parser<ValueCalculator> UnquotedValueContents =
+            InterpolatedValue
+                .Or(from inlineWhitespaces in InlineWhitespace
+                    from _ in Parse.Char('#').Not() // "#" after a whitespace is the beginning of a comment --> not allowed
+                    from partOfValue in NotControlNorWhitespace("$\"'") // quotes are not allowed in values, because in a shell they lead to inconsistent results
```

Comment:
They are not "inconsistent results" they are "different results than what they would mean here". Please use language closer to mine

Reply:
Updated, should be okay now I guess.